src/jobrunner.rs

```diff
@@ -40,15 +43,16 @@ struct JobRunner {
 
 impl JobRunner {
     fn new(snare: Arc<Snare>) -> Result<Self, Box<dyn Error>> {
-        assert!(snare.config.maxjobs < std::usize::MAX);
-        let mut running = Vec::with_capacity(snare.config.maxjobs);
-        running.resize_with(snare.config.maxjobs, || None);
-        let mut pollfds = Vec::with_capacity(snare.config.maxjobs * 2 + 1);
-        pollfds.resize_with(snare.config.maxjobs * 2 + 1, || {
-            PollFd::new(-1, PollFlags::empty())
-        });
+        // If the unwrap() on the lock fails, the other thread has paniced.
+        let maxjobs = snare.config.lock().unwrap().maxjobs;
+        assert!(maxjobs < std::usize::MAX);
+        let mut running = Vec::with_capacity(maxjobs);
+        running.resize_with(maxjobs, || None);
+        let mut pollfds = Vec::with_capacity(maxjobs * 2 + 1);
```

Comment:
Perhaps the above assertion is to cater for the `+ 1` here?

Reply:
Correct.